external/header.php

```diff
@@ -112,17 +113,17 @@ public static function __callstatic($name, $arguments)
 unset($domain);
 
 //Display warning if extension not available
-if (extension_loaded('xhprof') && $_xhprof['doprofile'] === true) {
+if (( extension_loaded('tideways') || extension_loaded('xhprof') ) && $_xhprof['doprofile'] === true) {
     include_once dirname(__FILE__) . '/../xhprof_lib/utils/xhprof_lib.php';
     include_once dirname(__FILE__) . '/../xhprof_lib/utils/xhprof_runs.php';
     if (isset($ignoredFunctions) && is_array($ignoredFunctions) && !empty($ignoredFunctions)) {
-        xhprof_enable(XHPROF_FLAGS_CPU + XHPROF_FLAGS_MEMORY, array('ignored_functions' => $ignoredFunctions));
+        (extension_loaded('tideways') ? tideways_enable(TIDEWAYS_FLAGS_CPU + TIDEWAYS_FLAGS_MEMORY, array('ignored_functions' => $ignoredFunctions)) : xhprof_enable(XHPROF_FLAGS_CPU + XHPROF_FLAGS_MEMORY, array('ignored_functions' => $ignoredFunctions)) );
```

Comment:
Although it works it does also look ugly. I would rather prefer this implemented as plugin system, where:

1. there will be plugin classes for each extension (xhprof, tideways, uprofiler)
2. each plugin will have method to: start/stop profiling
3. the plugin manager will determine if one of plugin extension is even loaded and if 0 OR more then 1 are loaded throw an exception